app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -0,0 +1,75 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.widget.Button;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.OnFocusChange;
+import butterknife.OnTextChanged;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+
+public class EntryNameActivity extends BaseMvpActivity<Presenter> implements View {
+
+    @BindView(R.id.contestant_welcome_text_view)
+    TextView contestantWelcomeTextView;
+    @BindView(R.id.entry_name_next_button)
+    Button entryNameNextButton;
+
+    @NonNull
+    @Override
+    public Presenter createPresenter(PresenterConfiguration configuration) {
+        return new EntryNamePresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_entry_name;
+    }
+
+    @Override
+    protected void onViewCreated(Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+    }
+
+    @Override
+    public void onContestNameRetrieved(String contestName) {
```

Comment:
Since this is something we're calling from the presenter I think it will be clearer if we name this setContestName, this way its more clear that the action will be on the presenter end.